modelscope/models/nlp/chatglm2/text_generation.py

```diff
@@ -1157,12 +1247,13 @@ def stream_chat(self,
                 outputs, past_key_values = outputs
             outputs = outputs.tolist()[0][len(inputs['input_ids'][0]):]
             response = tokenizer.decode(outputs)
-            response = self.process_response(response)
-            new_history = history + [(query, response)]
-            if return_past_key_values:
-                yield response, new_history, past_key_values
-            else:
-                yield response, new_history
+            if response and response[-1] != '�':
```

Comment:
define special token in a constant file, do not hard code here

Reply:
This code is from the official repo, it's not easy to maintain this code if we put this in the constant file when we update this code in the future.